### PR TITLE
hiop: fix cuda constraints and add tag to versions

### PR DESCRIPTION
**Pull request by cameronrutherford**
cc @cnpetra @pelesh @ryandanehy

### CODE REVIEW

**Comment by cameronrutherford** [comment]
cc @nychiang

**Comment by cameronrutherford** [comment]
@pelesh or @ryandanehy can you please review/approve this?

@nychiang or @cnpetra your approval would be helpful as well

File: var/spack/repos/builtin/packages/hiop/package.py [hunk 1]
```diff
@@ -22,33 +22,43 @@ class Hiop(CMakePackage, CudaPackage, ROCmPackage):
...
+    version(
+        "0.7.2", tag="v0.7.2", commit="d0f57c880d4202a72c62dd1f5c92e3bc8acb9788", submodules=True
+    )
+    version(
+        "0.7.1", tag="v0.7.1", commit="8064ef6b2249ad2feca92a9d1e90060bad3eebc7", submodules=True
+    )
+    version(
+        "0.7.0", tag="v0.7.0", commit="5f42ab34b419b7cf64d0fffb29d443b009dbfd75", submodules=True
+    )
+    version("0.6.2", tag="v0.6.2", commit="55652fbe923ab9107d002d0d070865bd22375b28")
```
Comment (nychiang) [inline var/spack/repos/builtin/packages/hiop/package.py:40]: @cameronrutherford I prefer the single-line representation, e.g., <=version("0.6.2")

**Review by nychiang — APPROVED** [review]
just one minor comment

File: var/spack/repos/builtin/packages/hiop/package.py (hunk 1 above)
Comment (cameronrutherford) [reply var/spack/repos/builtin/packages/hiop/package.py:40]: Unfortunately `spack style` invalidates any opinions you might have :(

**Review by eugeneswalker — APPROVED** [review]

**Comment by eugeneswalker** [comment]
Should we wait for additional review / comments or proceed now with merge?

**Comment by cameronrutherford** [comment]
Good to merge

**Pull request merged**